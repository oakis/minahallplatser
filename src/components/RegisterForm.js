import React, { Component } from 'react';
import { Text, Dimensions } from 'react-native';
import { Container, Content, Button, Input, InputGroup, Spinner } from 'native-base';
import { connect } from 'react-redux';
import {
	emailChanged,
	passwordChanged,
	passwordSecondChanged,
	registerUser,
	resetRoute
} from '../actions';
import minahallplatser from '../themes/minahallplatser';

class RegisterForm extends Component {
	
	componentWillMount() {
		this.props.resetRoute();
	}

	componentWillUnmount() {
		this.props.resetRoute();
	}

	onEmailChange(text) {
		this.props.emailChanged(text);
	}

	onPasswordChange(text) {
		this.props.passwordChanged(text);
	}

	onPasswordSecondChange(text) {
		this.props.passwordSecondChanged(text);
	}

	onButtonPress() {
		this.props.loading = true;
		const { email, password, passwordSecond } = this.props;
		this.props.registerUser({ email, password, passwordSecond });
	}

	renderSpinner() {
		if (this.props.loading) {
			return <Spinner color="#fff" />;
		}

		return 'Registrera';
	}

	render() {
		const width = Dimensions.get('window').width * 0.8;
		const styles = {
			errorStyle: {
				fontSize: 20,
				textAlign: 'center',
				color: 'red',
				width
			}
		};
		return (
			<Container>
				<Content
					contentContainerStyle={{
						flex: 1,
						flexDirection: 'column',
						justifyContent: 'flex-start',
						alignItems: 'center'
					}}
					keyboardShouldPersistTaps="always"
					keyboardDismissMode="on-drag"
				>
					<InputGroup borderType="underline" style={{ width }}>
						<Input
							placeholder="din@email.se"
							label="Email"
							keyboardType="email-address"
							autoFocus
							returnKeyType="next"
							onChangeText={this.onEmailChange.bind(this)}
							value={this.props.email}
						/>
					</InputGroup>

					<InputGroup borderType="underline" style={{ width }}>
						<Input
							secureTextEntry
							placeholder="ditt lösenord"
							label="Lösenord"
							onChangeText={this.onPasswordChange.bind(this)}
							value={this.props.password}
						/>
					</InputGroup>

					<InputGroup borderType="underline" style={{ width }}>
						<Input
							secureTextEntry
							placeholder="lösenord igen"
							label="Lösenord"
							onChangeText={this.onPasswordSecondChange.bind(this)}
							value={this.props.passwordSecond}
						/>
					</InputGroup>

					<Button
						primary
						block
						capitalize
						onPress={this.onButtonPress.bind(this)}
						style={{ width, marginTop: 10, alignSelf: 'center' }}
					>
						{this.renderSpinner()}
					</Button>

					<Text style={styles.errorStyle}>
						{this.props.error}
					</Text>

				</Content>
			</Container>
		);
	}

}

const MapStateToProps = (state) => {
	const { error, loading, email, password, passwordSecond } = state.auth;
	return { error, loading, email, password, passwordSecond };
};

export default connect(MapStateToProps,
	{ emailChanged, passwordChanged, passwordSecondChanged, registerUser, resetRoute }
)(RegisterForm);
