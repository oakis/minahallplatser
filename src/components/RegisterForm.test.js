import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import RegisterForm from './RegisterForm';


const mockStore = configureMockStore([thunk]);

const initialState = {
    auth: {
        email: 'abc@123.com',
        password: 'abc123',
        passwordSecond: 'abc123',
        loading: false,
        loadingFacebook: false,

    },
    errors: {
        error: 'An error message',
    },
    fav: {
        favorites: [],
        lines: [],
    }
};

it('should match snapshot', () => {
    const wrapper = shallow(<RegisterForm store={mockStore(initialState)} resetRoute={jest.fn()} />).dive();
    expect(wrapper).toMatchSnapshot();
});
