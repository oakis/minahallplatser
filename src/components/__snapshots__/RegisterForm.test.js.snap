// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegisterForm
    email="abc@123.com"
    error="An error message"
    loading={false}
    loadingFacebook={false}
    password="abc123"
    passwordSecond="abc123"
    resetRoute={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Message
          message="An error message"
          type="danger"
        />,
        <Input
          icon="ios-mail"
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="din@email.se"
          returnKeyType="next"
          value="abc@123.com"
        />,
        <Input
          icon="ios-key"
          iconSize={22}
          onChangeText={[Function]}
          placeholder="ditt lösenord"
          secureTextEntry={true}
          value="abc123"
        />,
        <Input
          icon="ios-key"
          iconSize={22}
          onChangeText={[Function]}
          placeholder="lösenord igen"
          secureTextEntry={true}
          value="abc123"
        />,
        <Button
          color="primary"
          label="Registrera"
          loading={false}
          onPress={[Function]}
          uppercase={true}
        />,
        <Button
          color="facebook"
          label="Registrera med facebook"
          loading={false}
          onPress={[Function]}
          uppercase={true}
        />,
        null,
      ],
      "keyboardDismissMode": "on-drag",
      "keyboardShouldPersistTaps": "always",
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginLeft": 10,
        "marginRight": 10,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "message": "An error message",
          "type": "danger",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "ios-mail",
          "keyboardType": "email-address",
          "onChangeText": [Function],
          "placeholder": "din@email.se",
          "returnKeyType": "next",
          "value": "abc@123.com",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "ios-key",
          "iconSize": 22,
          "onChangeText": [Function],
          "placeholder": "ditt lösenord",
          "secureTextEntry": true,
          "value": "abc123",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "ios-key",
          "iconSize": 22,
          "onChangeText": [Function],
          "placeholder": "lösenord igen",
          "secureTextEntry": true,
          "value": "abc123",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "primary",
          "label": "Registrera",
          "loading": false,
          "onPress": [Function],
          "uppercase": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "facebook",
          "label": "Registrera med facebook",
          "loading": false,
          "onPress": [Function],
          "uppercase": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Message
            message="An error message"
            type="danger"
          />,
          <Input
            icon="ios-mail"
            keyboardType="email-address"
            onChangeText={[Function]}
            placeholder="din@email.se"
            returnKeyType="next"
            value="abc@123.com"
          />,
          <Input
            icon="ios-key"
            iconSize={22}
            onChangeText={[Function]}
            placeholder="ditt lösenord"
            secureTextEntry={true}
            value="abc123"
          />,
          <Input
            icon="ios-key"
            iconSize={22}
            onChangeText={[Function]}
            placeholder="lösenord igen"
            secureTextEntry={true}
            value="abc123"
          />,
          <Button
            color="primary"
            label="Registrera"
            loading={false}
            onPress={[Function]}
            uppercase={true}
          />,
          <Button
            color="facebook"
            label="Registrera med facebook"
            loading={false}
            onPress={[Function]}
            uppercase={true}
          />,
          null,
        ],
        "keyboardDismissMode": "on-drag",
        "keyboardShouldPersistTaps": "always",
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 10,
          "marginRight": 10,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "message": "An error message",
            "type": "danger",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "ios-mail",
            "keyboardType": "email-address",
            "onChangeText": [Function],
            "placeholder": "din@email.se",
            "returnKeyType": "next",
            "value": "abc@123.com",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "ios-key",
            "iconSize": 22,
            "onChangeText": [Function],
            "placeholder": "ditt lösenord",
            "secureTextEntry": true,
            "value": "abc123",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "ios-key",
            "iconSize": 22,
            "onChangeText": [Function],
            "placeholder": "lösenord igen",
            "secureTextEntry": true,
            "value": "abc123",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "primary",
            "label": "Registrera",
            "loading": false,
            "onPress": [Function],
            "uppercase": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "facebook",
            "label": "Registrera med facebook",
            "loading": false,
            "onPress": [Function],
            "uppercase": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
